rts/System/Sound/OpenAL/MusicStream.cpp

```diff
@@ -1,36 +1,19 @@
 /* This file is part of the Spring engine (GPL v2 or later), see LICENSE.html */
 #include "MusicStream.h"
 
-#include "System/FileSystem/FileHandler.h"
 #include "System/Sound/SoundLog.h"
 #include "System/SafeUtil.h"
 #include "ALShared.h"
 #include "VorbisShared.h"
-#include "System/Sound/OpenAL/OggDecoder.h"
-#include "System/Sound/OpenAL/Mp3Decoder.h"
 
-#include <cstring> //memset
 #include <filesystem>
 #include <string_view>
 #include <variant>
 
-// NOTE:
-//   this buffer gets recycled by each new stream, across *all* audio-channels
-//   as a result streams are limited to only ever being played within a single
-//   channel (currently BGMusic), but cause far less memory fragmentation
-// TODO:
-//   can easily be fixed if necessary by giving each channel its own index and
-//   passing that along to the callbacks via COggStream{::Play}
-// CFileHandler fileBuffers[NUM_AUDIO_CHANNELS];
-static CFileHandler fileBuffer("", "");
 
-// FIXME these parts of MusicStream are outside the class because every instance
-// of CSoundSource has a copy of MusicStream class but only one is ever active
-static std::variant<OggDecoder, Mp3Decoder> decoder;
-
-
-MusicStream::MusicStream(ALuint _source)
-	: source(_source)
+MusicStream::MusicStream()
+	: buffers{}
```

Comment:
There's `std::fill` left in the constructor body, should be redundant with `buffers{}`